cpp/include/cudf/detail/search.hpp

```diff
@@ -105,12 +105,29 @@ rmm::device_uvector<bool> contains(
  * This function is designed for nested types only. It can also work with non-nested types
  * but with lower performance due to the complexity of the implementation.
  *
- * @param haystack The column containing search space.
- * @param needle A scalar value to check for existence in the search space.
- * @return true if the given `needle` value exists in the `haystack` column.
+ * @param haystack The column containing search space
+ * @param needle A scalar value to check for existence in the search space
+ * @return true if the given `needle` value exists in the `haystack` column
  */
 bool contains_nested_element(column_view const& haystack,
                              column_view const& needle,
                              rmm::cuda_stream_view stream);
```

Comment:
Can we use `is_nested` trait to simplify the logic thus this function is an `if` branch in the implementation of `contains`? https://github.com/rapidsai/cudf/blob/ed9355ff1321770225efc073bf2ee0e81c7db0d8/cpp/include/cudf/detail/search.hpp#L52-L57

This will help get rid of the odd `contains_nested_element` and `multi_contains_nested_elements` naming. e.g. The ovarload of `contains` returning a column of BOOL is not named as `multi_contains`.

For me, `contains_nested_element` takes some data and returns a boolean indicating if nested elements are present in that data, which is apparently not the semantic here.

Reply:
Sure. These functions were extracted to improve compile time with nested structs. Now we can unify them back to `contains_*_dispatch`.

Reply:
`contains_scalar` is much cleaner! Thanks @ttnghia !